### PR TITLE
Effect Firetail OK

### PR DESCRIPTION
**Pull request by hensldm**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c [hunk 1]
```diff
@@ -1444,7 +1444,7 @@ void EnBigpo_DrawCircleFlames(Actor* thisx, PlayState* play) {
 
         gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-        gSPDisplayList(POLY_XLU_DISP++, &gGameplayKeepDrawFlameDL);
+        gSPDisplayList(POLY_XLU_DISP++, &gEffFire1DL);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c:1447]:
```suggestion
        gSPDisplayList(POLY_XLU_DISP++, gEffFire1DL);
```

File: src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c [hunk 2]
```diff
@@ -1471,7 +1471,7 @@ void EnBigpo_RevealedFire(Actor* thisx, PlayState* play) {
 
     gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-    gSPDisplayList(POLY_XLU_DISP++, &gGameplayKeepDrawFlameDL);
+    gSPDisplayList(POLY_XLU_DISP++, &gEffFire1DL);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c:1474]:
```suggestion
    gSPDisplayList(POLY_XLU_DISP++, gEffFire1DL);
```

File: src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c [hunk 3]
```diff
@@ -429,7 +429,7 @@ void ObjChan_DrawFire(ObjChan* this, PlayState* play) {
     gSPSegment(&dl[1], 0x08, Gfx_TwoTexScroll(play->state.gfxCtx, 0, 0, 0, 32, 64, 1, 0, -sp4C * 20, 32, 128));
     gDPSetPrimColor(&dl[2], 128, 128, 255, 255, 0, 255);
     gDPSetEnvColor(&dl[3], 255, 0, 0, 0);
-    gSPDisplayList(&dl[4], &gGameplayKeepDrawFlameDL);
+    gSPDisplayList(&dl[4], &gEffFire1DL);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c:432]:
```suggestion
    gSPDisplayList(&dl[4], gEffFire1DL);
```

File: src/overlays/effects/ovl_Effect_Ss_Fire_Tail/z_eff_ss_fire_tail.c [hunk 4]
```diff
@@ -1,27 +1,142 @@
...
+            this->pos.z =
+                player->bodyPartsPos[bodyPart].z - (Math_CosS(Camera_GetCamDirYaw(GET_ACTIVE_CAM(play))) * 5.0f);
+
+            Matrix_Translate(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+        }
+    } else {
+        Matrix_Translate(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+    }
+
+    yaw = Math_Vec3f_Yaw(&scale, &this->vec) - Camera_GetCamDirYaw(GET_ACTIVE_CAM(play));
```
Comment (engineer124) [inline src/overlays/effects/ovl_Effect_Ss_Fire_Tail/z_eff_ss_fire_tail.c:105]:
```suggestion
    yawDiff = Math_Vec3f_Yaw(&scale, &this->vec) - Camera_GetCamDirYaw(GET_ACTIVE_CAM(play));
```
Perhaps?

File: src/overlays/effects/ovl_Effect_Ss_Fire_Tail/z_eff_ss_fire_tail.c [hunk 5]
```diff
@@ -1,27 +1,142 @@
...
+    this->rPrimColorR = initParams->primColor.r;
+    this->rPrimColorG = initParams->primColor.g;
+    this->rPrimColorB = initParams->primColor.b;
+    this->rEnvColorR = initParams->envColor.r;
+    this->rEnvColorG = initParams->envColor.g;
+    this->rEnvColorB = initParams->envColor.b;
+    this->rBodyPart = initParams->bodyPart;
+    this->rType = initParams->type;
+
+    return 1;
```
Comment (isghj5) [inline src/overlays/effects/ovl_Effect_Ss_Fire_Tail/z_eff_ss_fire_tail.c:67]: er.. I forget is this bool?

Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Fire_Tail/z_eff_ss_fire_tail.c:67]: Probably, but just haven't started doing it yet

**Review by EllipticEllipsis — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**